packages/config/src/configDefaults.ts

```diff
@@ -2,6 +2,12 @@ import * as path from "path";
 import Provider from "@truffle/provider";
 import TruffleConfig from "./";
 
+let provider: any = null;
+
+export const resetProvider = () => {
+  provider = null;
+};
+
```

Comment:
I believe a test like this would now fail:

```
const config1 = Config.load(pathToConfig);
const config2 = Config.load(pathToOtherConfig);
assert.notStrictEqual(config1.networks.develop.provider, config2.networks.develop.provider); // this fails, no?
```

I think we should treat Truffle as a whole as a library, not only as a CLI tool with short lived commands. `@truffle/config` _is_ a major versioned (stable API) library now (with 31000+ weekly downloads): https://www.npmjs.com/package/@truffle/config

I think the memoization needs to be solved another way.

Reply:
Can you elaborate a bit further? I'm not sure I'm following you here, why would that fail and what does that have to do with your comments about @truffle/config having a lot of downloads and treating Truffle as a whole library?

Reply:
This is great and I think it's the exact kind of insight we need from the team suss these things out. I will make some additional unit tests to cover these cases and it would be helpful to keep addressing them as this saga continues. 

Since a lot of these libs were built organically and issues were addressed as they came in, this would really help us get a bigger picture of the tradeoffs so we can perhaps formalize something. E.g. 1 dev did _this_ and another did _that_ and now to a change to this impacts that, or vice versa and it reaches deadlock.

I'm convinced that this is impossible to solve with a silver bullet that satisfies every implementation that is in place and we'll have to get the big picture if we want to solve this problem.

Reply:
Ok, is that a problem?

Reply:
Yea, it would be bad. David raised a good issue. I'm personally guilty, of thinking `@truffle/config` is an internal project.